www-apps/gitea/gitea-1.21.4.ebuild

```diff
@@ -0,0 +1,151 @@
+
+	gitea_settings=(
+		"-X code.gitea.io/gitea/modules/setting.CustomConf=${EPREFIX}/etc/gitea/app.ini"
+		"-X code.gitea.io/gitea/modules/setting.CustomPath=${EPREFIX}/var/lib/gitea/custom"
+		"-X code.gitea.io/gitea/modules/setting.AppWorkPath=${EPREFIX}/var/lib/gitea"
+	)
+
+	makeenv=(
+		LDFLAGS="-extldflags \"${LDFLAGS}\" ${gitea_settings[*]}"
+		TAGS="${gitea_tags}"
+	)
+
+	if [[ ${PV} != *9999 ]]; then
+		# Use variable STORED_VERSION_FILE (the "${S}/VERSION" file) to set version,
+		# and prevent executing git command when it's not a live version.
+		makeenv+=( GITHUB_REF_NAME="" )
+	fi
+
+	if use pie ; then
+		# Please check the supported platforms when a new keyword request opened,
+		# refer to file: 'go/src/internal/platform/supported.go'.
+		# When PIE buildmode is not supported by internal linker, the external
+		# linker will be used automatically, refer to:
+		# https://github.com/golang/go/blob/ed817f1c4055a559a94afffecbb91c78e4f39942/src/cmd/link/internal/ld/config.go#L149
+		makeenv+=( EXTRA_GOFLAGS="-buildmode=pie" )
+	fi
+
+	env "${makeenv[@]}" emake backend
+}
+
+src_install() {
+	dobin gitea
+
+	einstalldocs
+
+	newconfd "${FILESDIR}/gitea.confd-r1" gitea
+	newinitd "${FILESDIR}/gitea.initd-r3" gitea
+	newtmpfiles - gitea.conf <<-EOF
+		d /run/gitea 0755 git git
+	EOF
+	systemd_newunit "${FILESDIR}"/gitea.service-r4 gitea.service
+
+	insinto /etc/gitea
+	newins custom/conf/app.example.ini app.ini
+	if use acct; then
+		fowners root:git /etc/gitea/{,app.ini}
+		fperms g+w,o-rwx /etc/gitea/{,app.ini}
+
+		diropts -m0750 -o git -g git
+		keepdir /var/lib/gitea /var/lib/gitea/custom /var/lib/gitea/data
+		keepdir /var/log/gitea
+	fi
+}
+
+pkg_postinst() {
+	fcaps_pkg_postinst
+
+	# It is not guaranteed that the git user and group always exist (due to the acct USE Flag),
+	# but for convenience, the tmpfile uses the git user and group by default.
+	# To avoid installation errors, a condition needs to be added here:
+	#   if there is no git user or group, the installation of tmpfile will be skipped
+	#   and the user will be notified to handle it by themselves.
+	if egetent passwd git &>/dev/null && \
+		egetent group git &>/dev/null; then
+		tmpfiles_process gitea.conf
+	else
+		eerror "Unable to install the tmpfile for gitea due to the git user or group is missing,"
+		eerror "please install tmpfile manually or rebuild this package with USE flag 'acct'."
+		eerror "You can simply copy the default tmpfile from '/usr/lib/tmpfiles.d/gitea.conf'"
+		eerror "to higher priority path '/etc/tmpfiles.d/gitea.conf', and correct it with"
+		eerror "the right User and Group value (see tmpfiles.d(5) for details), then execute:"
+		eerror "  # systemd-tmpfiles --create /etc/tmpfiles.d/gitea.conf"
+		eerror "to install it."
+	fi
+
+	if [[ -n ${REPLACING_VERSIONS} ]] ; then
+		local -i major minor
+		IFS="." read -r major minor _ <<<"$REPLACING_VERSIONS"
+		if (( major == 1 && minor < 21 )); then
```

Comment:
this is kind of tedious.. can you take a look at 'version_is_at_least' (or version_compare, in versionator.eclass)? it should work for you

Reply:
`versionator.eclass` does not support EAPI 8, use `ver_test`, see https://mgorny.pl/articles/the-ultimate-guide-to-eapi-7.html#version-comparison-ver-test for more information.

Reply:
Corrected, thank you very much for pointing this out!